spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -268,29 +274,40 @@ public void stop(Runnable callback) {
 				new KafkaMessageDrivenChannelAdapter<>(
 						messageListenerContainer);
 		kafkaMessageDrivenChannelAdapter.setBeanFactory(this.getBeanFactory());
-		final RetryTemplate retryTemplate = buildRetryTemplate(extendedConsumerProperties);
-		kafkaMessageDrivenChannelAdapter.setRetryTemplate(retryTemplate);
+		if (extendedConsumerProperties.getMaxAttempts() > 1) {
```

Comment:
Wondering if in a next pass we could move all up to `AbstractMessageChannelBinder` assuming that the ChannelAdapter used for a binder accepts a recoveryCallback and an ErrorChannel ?